modin/pandas/test/test_io.py

```diff
@@ -1453,6 +1454,65 @@ def comparator(df1, df2):
                 comparator=comparator,
             )
 
+    @pytest.mark.parametrize("columns", [None, ["col1"]])
+    @pytest.mark.parametrize(
+        "filters",
+        [None, [("col1", "<=", 1_000_000)], [("col1", "<=", 75), ("col2", ">=", 35)]],
+    )
+    @pytest.mark.parametrize(
+        "range_index_start",
+        [0, 5_000],
+    )
+    @pytest.mark.parametrize(
+        "range_index_step",
+        [1, 10],
+    )
+    @pytest.mark.parametrize(
+        "range_index_name",
+        [None, "my_index"],
+    )
+    def test_read_parquet_index(
+        self,
+        engine,
+        make_parquet_file,
+        columns,
+        filters,
+        range_index_start,
+        range_index_step,
+        range_index_name,
+    ):
+        if engine == "pyarrow" and filters == []:
+            # pyarrow, and therefore pandas using pyarrow, errors in this case.
+            # Modin correctly replicates this behavior; however error cases
+            # cause race conditions with ensure_clean on Windows.
+            # TODO: Remove this once https://github.com/modin-project/modin/issues/6460 is fixed.
+            pytest.xfail(
+                "Skipping empty filters error case to avoid race condition - see #6460"
+            )
```

Comment:
Can we `.xfail` it on Windows only, to make sure code is still tested on Linux?

Reply:
This code comes from the previous test. Your comment does make me realize that I could have factored out a helper function for these two tests, since they are the same thing but `@pytest.mark.parametrize`'d differently. I will factor it out, and then make the improvement you suggest, since I think it is a good one and I'll be touching the existing code anyway.